addons/website_event_track/static/src/js/event_track_reminder.js

```diff
@@ -39,46 +41,133 @@ publicWidget.registry.websiteEventTrackReminder = publicWidget.Widget.extend({
-                        'add_track_to_favorite',
-                        {
-                            title: _t('Allow push notifications?'),
-                            body: _t('You have to enable push notifications to get reminders for your favorite tracks.'),
-                            delay: 0
-                        },
-                    ]);
-                }
+                Component.env.bus.trigger('open_notification_request', [
+                    'add_track_to_favorite',
+                    {
+                        title: _t('Allow push notifications?'),
+                        body: _t('You have to enable push notifications to get reminders for your favorite tracks.'),
+                        delay: 0
+                    },
+                ]);
             }
         });
     },
 
+    _removeReminder: function (trackId) {
+        var self = this;
+        rpc('/event/track/toggle_reminder', {
+            track_id: trackId,
+            set_reminder_on: false,
+        }).then(function (result) {
+            self.$('.o_wetrack_js_reminder_text').text(_t('Set Favorite'));
+            self.reminderOn = false;
+            self._updateDisplay();
+            self.notification.add(_t('Talk removed from your Favorites'), {
+                type: 'info',
+                sticky: true
+            });
+        });
+    },
+
+    _validateAddReminder: function (trackId) {
+        this._addReminder(trackId);
+        this.notification.add(_t('Track successfully added to your favorites. Check your email to add them to your agenda.'),
+            {
+                type: 'info',
+                sticky: true,
+                className: 'o_send_email_reminder_success'
+            });
+    },
+
+    _sendEmailReminder: function (trackId) {
+        rpc('/event/send_email_reminder',  {
+            track_id: trackId,
+        }).then( (result) => {
+            this._validateAddReminder(trackId);
+        });
+    },
+
+    _displayModalEmailReminder: function (trackId) {
+        rpc('/event/display_modal_email_reminder', {
+            'track_id': trackId,
+        }).then( (result) => {
+            var self = this;
+            $('body').prepend(result);
+            var modalEmailReminder = $('.o_wetrack_js_modal_email_reminder');
+            modalEmailReminder.find(".o_form_button_cancel").on('click', function () {
+                modalEmailReminder.remove();
+            });
+            modalEmailReminder.on('submit', function (ev) {
+                ev.preventDefault();
+                var data = Object.fromEntries(new FormData(ev.target).entries());
+                rpc('/event/add_session_email_reminder', data).then( (result) => {
+                    self._validateAddReminder(trackId);
+                });
+                modalEmailReminder.remove();
+            });
+        });
+    },
+
+    _checkEmailReminder: function (trackId){
+        rpc('/event/has_email_reminder').then( (result) => {
```

Comment:
The rpc is still used because it does not seem possible to catch session keys added from the controller. So it's not possible to know if the visitor already submits an email address. Maybe there is a way to avoid this rpc but I do not see it.